plutus-benchmark/lists/test/Sum/right-fold-built-in.budget.golden

```diff
@@ -1,2 +1,2 @@
-({cpu: 167459654
-| mem: 534532})
\ No newline at end of file
+({cpu: 181259654
```

Comment:
interesting that many of these are not better at all!

Reply:
My guess, without looking at the PIR: `Strict` causes GHC to inline fewer things, which puts more burden on our own inliner. Once we get #5308 and #5323 in, things may improve dramatically.

Evidence: https://github.com/input-output-hk/plutus/pull/5292#discussion_r1184733507

Reply:
I think we should check out the optimized PIR here before merging.

Reply:
Yeah and I'll revisit this after merging #5308 and #5323